src/order/filter/countable_Inter.lean

```diff
@@ -189,3 +189,48 @@ begin
   refine ⟨λ S hSc hS, ⟨_, _⟩⟩; refine (countable_sInter_mem hSc).2 (λ s hs, _),
   exacts [(hS s hs).1, (hS s hs).2]
 end
+
+namespace filter
+
+variable (g : set (set α))
+
+/-- `countable_generate_sets g` is the (sets of the)
+smallest `countable_Inter_filter` containing `g`.-/
+inductive countable_generate_sets : set α → Prop
+| basic {s : set α}      : s ∈ g → countable_generate_sets s
+| univ                   : countable_generate_sets univ
+| superset {s t : set α} : countable_generate_sets s → s ⊆ t → countable_generate_sets t
+| Inter {S : set (set α)}  : S.countable →
+    (∀ s ∈ S, countable_generate_sets s) → countable_generate_sets ⋂₀ S
+
+/-- `countable_generate g` is the smallest `countable_Inter_filter` containing `g`.-/
```

Comment:
Can you add a proof that this is true?

Reply:
sure, although I guess it should be "biggest" with the way the filter order works.

Reply:
done.

Reply:
Sorry, what I meant is to show that `countable_generate g = ⨅ f [countable_Inter_filter f],  g ≤ f`

Reply:
Is the way I phrased it now fine?

```
lemma countable_generate_is_greatest : is_greatest
  {f : filter α | countable_Inter_filter f ∧ g ⊆ f.sets} (countable_generate g)
```